pkg/cmd/network.go

```diff
@@ -16,12 +17,24 @@ func withSubnetCIDRFlags(flags []cli.Flag) []cli.Flag {
 			Usage:  "IP address range for Pods",
 			Value:  k0sv1beta1.DefaultNetwork().PodCIDR,
 			Hidden: true,
+			Action: func(c *cli.Context, addr string) error {
+				if _, _, err := net.ParseCIDR(addr); err != nil {
+					return fmt.Errorf("unable to parse pod cidr: %w", err)
+				}
+				return nil
+			},
 		},
 		&cli.StringFlag{
 			Name:   "service-cidr",
 			Usage:  "IP address range for Services",
 			Value:  k0sv1beta1.DefaultNetwork().ServiceCIDR,
 			Hidden: true,
+			Action: func(c *cli.Context, addr string) error {
+				if _, _, err := net.ParseCIDR(addr); err != nil {
```

Comment:
should these use the same netutils code as the cidr flag?
maybe with a /17 requirement instead of /16

Reply:
I don't think we want to restrict this too much. I am worried with people that may be already using this with a `/24` or something. As this is now a hidden flag we may consider printing a warning here but I would argue this should be part of a different issue.